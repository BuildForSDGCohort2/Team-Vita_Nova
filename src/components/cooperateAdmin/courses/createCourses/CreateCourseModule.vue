<template>
  <v-container>
    <v-responsive>
      <h3 class="ma-4 " style="padding-top: 10%">Untitled Module</h3>
      <v-card class="text-center justify-center ma-4">
        <v-container fluid>
          <v-text-field
            v-model="model"
            :items="items"
            :search-input.sync="search"
            hide-selected
            hint="Donâ€™t worry, you can always change this."
            label="Title"
            multiple
            outlined
            persistent-hint
            chips
            color="red"
            class=" title-box "
          >
            <template v-slot:no-data>
              <v-list-item>
                <v-list-item-content>
                  <v-list-item-title>
                    No course matching "<strong>{{ search }}</strong
                    >". Press <kbd>enter</kbd> to create a new one
                  </v-list-item-title>
                </v-list-item-content>
              </v-list-item>
            </template>
          </v-text-field>
        </v-container>
      </v-card>

      <v-btn
        class="ma-4"
        @click="saveCourse"
        style="background-color: #FF2E2E; color: #F8F8F8; height: 50px !important; width: 120px !important;"
        >Save</v-btn
      >

      <v-card
        flat
        class="text-center justify-center ma-4"
        style="background: #F8F8F8;border-radius: 11.2578px; height: 60px; "
      >
        <v-container fluid>
          <h6 class="tip-text">
            Tip: Chapter(s) will be set to draft if there are no lessons created
            for it.
          </h6>
        </v-container>
      </v-card>
    </v-responsive>
  </v-container>
</template>

<script>
export default {
  name: "CreateCourseModule",

  data: () => ({
    items: ["Software Engineering", "Business Management", "Design Thinking"],
    model: [],
    search: null
  }),

  watch: {
    model(val) {
      if (val.length > 5) {
        this.$nextTick(() => this.model.pop());
      }
    }
  },

  methods: {
    saveCourse() {}
  }
};
</script>

<style scoped>
.title-box {
  width: 70%;
  margin-left: 15%;
}
.tip-text {
  font-family: IBM Plex Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 13.5094px;
  line-height: 18px;
  /* identical to box height */
  margin-right: 25%;
  margin-top: 5px;
  /* semicolon-neutral-black */

  color: #2b1c1c;
}
</style>
