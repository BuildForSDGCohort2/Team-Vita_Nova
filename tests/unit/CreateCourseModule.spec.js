import { mount } from "@vue/test-utils";
import CreateCourseModule from "@/components/cooperateAdmin/courses/createCourses/CreateCourseModule";

const wrapper = mount(CreateCourseModule);
//testing if the test is properly setup on the run
describe("CreateCourseModule", () => {
    it("setup correctly", () => {
        expect(true).toBe(true);
    });

    //test to check if the title is correct
    it("test page has title", () => {
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain("Untitled Module");
    });

    it('will change and update the value', () => {
        expect(wrapper.vm.accept).toBe(false)

        // const checkbox = wrapper.vm.$refs.checkbox
        // checkbox.$emit('input', true)
        //
        // expect(wrapper.vm.accept).toBe(true)
    });

    it("sets the correct default data", () => {
        expect(typeof CreateCourseModule.data).toBe("function");
        const defaultData = CreateCourseModule.data();
        expect(defaultData.search).toBe(null);
    });

});
