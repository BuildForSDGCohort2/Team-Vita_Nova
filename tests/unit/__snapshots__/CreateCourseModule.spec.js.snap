// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourseModule test page has title 1`] = `
<v-container>
  <v-responsive>
    <h3 class="ma-4 " style="padding-top: 5%;">Untitled Module</h3>
    <v-card class="ma-4">
      <validationobserver></validationobserver>
    </v-card>
    <section class="ma-4" style="padding-top: 100px; padding-bottom: 100px;">
      <h3>Lesson Settings</h3>
      <v-checkbox color="red"></v-checkbox>
      <v-checkbox color="red"></v-checkbox>
    </section>
    <v-card flat="" max-width="686" class="text-center justify-center ma-4" style="background: rgb(248, 248, 248); border-radius: 11.2578px; height: 60px;">
      <v-container fluid="">
        <h6 class="tip-text">
          Tip: Chapter(s) will be set to draft if there are no lessons created
          for it.
        </h6>
      </v-container>
    </v-card>
  </v-responsive>
</v-container>
`;
